xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -101,7 +104,10 @@ static String canonifyResourceName(String resourceName) {
     return resourceName.replace(rawQuery, canonifiedQuery);
   }
 
-  static String percentEncodePath(String input) {
+  /*
+   * Percent encode the input using the url path segment escaper.
+   */
+  public static String percentEncodePath(String input) {
```

Comment:
same here, can you add
`@ExperimentalApi("https://github.com/grpc/grpc-java/issues/10862")`

Reply:
Done